controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -61,5 +62,32 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, err
 	}
 
+	// add toolchaincluster role label if not present
+	reqLogger.Info("adding cluster role label based on type")
+	if err := r.addToolchainClusterRoleLabelFromType(reqLogger, toolchainCluster); err != nil {
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, r.clusterCacheService.AddOrUpdateToolchainCluster(toolchainCluster)
 }
+
+func (r *Reconciler) addToolchainClusterRoleLabelFromType(log logr.Logger, toolchainCluster *toolchainv1alpha1.ToolchainCluster) error {
+	if _, clusterTypeFound := toolchainCluster.Labels[cluster.LabelType]; !clusterTypeFound {
+		log.Info("cluster `type` label not found, unable to add toolchain cluster role label from type")
+		return nil
+	}
+	if toolchainCluster.Labels[cluster.LabelType] != string(cluster.Member) {
+		log.Info("cluster `type` is not member, skipping cluster role label setting")
+		return nil
+	}
+
+	clusterRoleLabel := cluster.ToolchainClusterRoleLabelTenant()
+	if _, exists := toolchainCluster.Labels[clusterRoleLabel]; !exists {
+		log.Info("setting cluster role label for toolchaincluster", clusterRoleLabel, toolchainCluster.Name)
+		toolchainCluster.Labels[clusterRoleLabel] = ""
```

Comment:
let's add a comment to explain that we only care about the label key, hence the value can remain empty.

Reply:
Makes sense! Thanks for the suggestion!

Reply:
added in [5fd587a46e83cad1](https://github.com/mfrancisc/toolchain-common/commit/5fd587a46e83cad164b50f71c52895b9fce79610)

Reply:
I cannot not see the comment in [5fd587a46e83cad1](https://github.com/mfrancisc/toolchain-common/commit/5fd587a46e83cad164b50f71c52895b9fce79610) :/ maybe I'm missing something?

Reply:
sorry, added it now! by mistake I've added it to cluster.go only https://github.com/mfrancisc/toolchain-common/blob/3f70ed9490d2ee884d730a8fe55976a5fdecaaef/pkg/test/verify/cluster.go#L216 🤦

Reply:
cool, thanks!